spec/models/api/hses_spec.rb

```diff
@@ -21,28 +21,86 @@
-    let(:issues) { Api::Hses::Issues.new(user: user) }
+    context "with params argument" do
+      let(:issues) {
+        Api::Hses::Issues.new(
+          user: user,
+          params: ActionController::Parameters.new({page: "2"})
+        )
+      }
 
-    describe "#initialize" do
-      it "sets @username" do
-        expect(issues.instance_variable_get(:@username)).to eq user["uid"]
+      describe "#initialize" do
+        it "sets @username" do
+          expect(issues.instance_variable_get(:@username)).to eq user["uid"]
+        end
+        it "sets @page" do
+          expect(issues.instance_variable_get(:@page)).to eq "2"
+        end
       end
-    end
 
-    describe "#host" do
-      it "returns the value set in the Api::Hses module" do
-        expect(issues.host).to eq Rails.configuration.x.hses.api_hostname
+      describe "#request" do
+        it "sends query parameters with correct offset for pagination" do
+          # stub the net/http request / response
+          response = Net::HTTPSuccess.new(1.0, "200", "OK")
+          expect_any_instance_of(Net::HTTP).to receive(:start).and_return response
+          expect(response).to receive(:body).and_return('{"meta":{},"data":[]}')
+
+          expect(URI).to receive(:encode_www_form).with(
+            {
+              types: 1,
+              username: "test.uid",
+              offset: 25,
+              limit: 25
+            }
+          )
+          issues.request
+        end
       end
     end
 
-    describe "#request" do
-      it "returns a Ruby object with meta and data keys" do
-        # stub the net/http request / response
-        response = Net::HTTPSuccess.new(1.0, "200", "OK")
-        expect_any_instance_of(Net::HTTP).to receive(:start).and_return response
-        expect(response).to receive(:body).and_return('{"meta":{},"data":[]}')
+    context "without params argument" do
+      let(:issues) { Api::Hses::Issues.new(user: user) }
+
+      describe "#initialize" do
+        it "sets @username" do
+          expect(issues.instance_variable_get(:@username)).to eq user["uid"]
+        end
+        it "sets @page to default value" do
+          expect(issues.instance_variable_get(:@page)).to eq 1
+        end
+      end
+
+      describe "#host" do
+        it "returns the value set in the Api::Hses module" do
+          expect(issues.host).to eq Rails.configuration.x.hses.api_hostname
+        end
+      end
+
+      describe "#request" do
+        it "returns a Ruby object with meta and data keys" do
+          # stub the net/http request / response
+          response = Net::HTTPSuccess.new(1.0, "200", "OK")
+          expect_any_instance_of(Net::HTTP).to receive(:start).and_return response
+          expect(response).to receive(:body).and_return('{"meta":{},"data":[]}')
+
+          expect(issues.request).to match({"meta" => {}, "data" => []})
+        end
+        it "sends query parameters for type, username, and pagination" do
```

Comment:
[annoying nitpick] i prefer a newline in between blocks to make it easier to read. thanks!